client/src/components/campaignCreatorComponents/TriggerEventsSettings.tsx

```diff
@@ -105,6 +111,40 @@ const CampaignFinalSettings = ({
         }, {});
     };
 
+    const createResourceGroup = (categoryKey: keyof TypeResourceListData) => {
+        return resourceListData[categoryKey].reduce((currentResourceObject, resource) => {
+            return {
+                ...currentResourceObject,
+                [resource]: 0,
+            };
+        }, {});
+    };
+
+    const createResourceList = () => {
+        // console.log('setttings', campaignSettings);
+        const monsterList = [
+            ...campaignSettings.node_nemesis_1,
+            ...campaignSettings.node_nemesis_2,
+            ...campaignSettings.node_nemesis_3,
+            ...campaignSettings.node_quarry_1,
+            ...campaignSettings.node_quarry_2,
+            ...campaignSettings.node_quarry_3,
+            ...campaignSettings.node_quarry_4,
+            campaignSettings.node_core,
+            'Basic',
+        ];
+        console.log('monster list', monsterList);
+        // return { ...createResourceGroup('Basic') };
+        return monsterList
+            .filter((monster) => Object.keys(resourceListData).includes(monster!))
+            .reduce((currentResourceObject, monster) => {
+                return {
+                    ...currentResourceObject,
+                    [monster!]: createResourceGroup(monster as keyof TypeResourceListData),
```

Comment:
Instead of `monster!` we should instead be doing `.filter((monster) => monster && Object.keys( [...]` methinks.

Reply:
That should resolve the typescript errors without presenting holes later in case `monster` does end up being null sometimes

Reply:
that works for line 139.
for line 143, i did the following:
`return monster
                    ? {
                          ...currentResourceObject,
                          [monster!]: createResourceGroup(monster as keyof TypeResourceListData),
                      }
                    : { ...currentResourceObject };`